Add storage_account parameter to Azure Blob Storage backend (#2016)
pulumi/pulumi-hugo: themes/default/content/docs/intro/concepts/state.md

- [Pulumi Self-Hosted Service](#logging-into-a-self-hosted-pulumi-service-backend)
- [Local Filesystem](#logging-into-the-local-filesystem-backend)
- [AWS S3 (or compatible server)](#logging-into-the-aws-s3-backend)

> - [Azure Blob Storage](#logging-into-the-azure-blob-storage-backend-using-the-env-vars)

Reviewer: Style: this heading should match the others, the S3 heading is an example of this

```suggestion
- [Azure Blob Storage](#logging-into-the-azure-blob-storage-backend)
```

$ pulumi login s3://<bucket-name>?endpoint=my.minio.local:8080&disableSSL=true&s3ForcePathStyle=true
```

> ##### Logging Into the Azure Blob Storage Backend using the env vars

Reviewer: Style, align heading with previous suggestion

```suggestion
##### Logging Into the Azure Blob Storage Backend
```

To tell Pulumi what Azure storage account to use, set the `AZURE_STORAGE_ACCOUNT` environment variable. Also, set either `AZURE_STORAGE_KEY` or `AZURE_STORAGE_SAS_TOKEN` to authorize access. For additional configuration options, see [Azure Setup]({{< relref "/registry/packages/azure/installation-configuration" >}}). If you're new to Azure Blob Storage, see [the Azure documentation](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-cli).

> ##### Logging Into the Azure Blob Storage Backend using az cli authentication (as of Pulumi 3.41.1)
> 
> Similarly, one can implicitly use az cli authentication by defining the storage account in the URL like so:
> 
> ```sh
> $ pulumi login azblob://<container-path>?storage_account=account_name
> ```

Reviewer: Style: modify this to match the AWS S3 example for adding additional query parameters.

````suggestion
{{% notes type="info"%}}
As of Pulumi CLI v3.41.1, instead of the environment variables above, the Azure CLI authentication may be used by specifying the storage account in the URL like so after using `az login`:

```sh
$ pulumi login azblob://<container-path>?storage_account=account_name
```

{{% /notes %}}
````

{{% notes type="info"%}}

> Make sure you have "Storage Blob Data Contributor" role assigned to you in IAM. Just having "Owner" is not enough, as that is a "management"-type of role.

Reviewer: Content: this would work with custom roles & with service accounts.

```suggestion
The Azure account must have the [Storage Blob Data Contributor role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor) or an equivalent role with permissions to read, write, and delete blobs.
```
